### DISCUSSION TITLE
feature_request: user input(intervention) during job running

### DISCUSSION THREAD

**Discussion by gh250100** [community/community#14230] [Actions] [answered]
Hello ,

it would be great if we could run workflow and during the run on specific job(stage) to have user input option ,
for example:

packer job that build image:
Job:
 get-files:
   1. job start with checkout 
   2. Script collect all FULLNAME files for *.pkr.hcl/json to /tmp/all-packer-templates.log  
   
 show-packer-templates:
   ONCE ACTION IS IN THIS STAGE SHOW USER 
   ALL PACKER TEMPLATES FROM /tmp/all-packer-templates.log  file
   USER SELECTED ONE OF THE OPTION 
   MY_TEMPLATE=USER SELECTED ONE OF THE OPTION 
   packer build .... $MY_TEMPLATE

short description - its like to use def function in jenkinsfile for that kind of use
![github-action](https://user-images.githubusercontent.com/67205688/161500060-ab07a29d-65d5-4cee-b1c2-5708168af1ed.PNG)

Thanks in advance 
Gil

**Comment by ethomson** [comment, marked as answer]
Thanks for the feedback! At the moment, you can use `workflow_dispatch` inputs to collect user data at the beginning of a run, but not _during_ a workflow run. I'll add this feedback to the backlog to consider how we might be able to support this scenario. Thanks again!

**Reply by elmijo** [reply]
Hey guys, I would like to say thanks to consider this feature, I'm a Jenkins user and I have to say, it is really useful, because it allows the user to choose which path their workflow will take, instead of creating complex logic. In theory it works quite similar to `workflow_dispatch`, with a difference from my point of view, it should have  a **timeout**, so,  if nobody fills/chooses/answers the input at a certain time, the process should be aborted or uses the default value.

I would like to share you a use case, publish an NPM package, we can use this feature to decide which version of our package we want to publish, push the new version to our package repository, create/push a new tag to Github and push`package.json`  changes   to main. 

Maybe you can say, hey! you can do that now using `workflow_dispatch`, and it's true but we should define a different workflow for that. This feature allow us to have everything together in the same workflow, and use it in the same flow whenever we need it with no extra processes 

BTW, In Jenkins, we have to do something like this to revolve the NPM publish use case:

```groovy
stage("Publish Package") {
  when {
    branch "main"
  }
  steps {
    script {          
      timeout(time: 30, unit: 'MINUTES') {
         env.RELEASE_SCOPE = input message: 'Do you want to publish?', ok: 'Publish',
           parameters: [choice(name: 'RELEASE_SCOPE', choices: 'patch\nminor\nmajor', description: 'What is the release scope?')]
      }
      withCredentials([gitUsernamePassword(credentialsId: '<my-github-secret>', gitToolName: 'git-tool')]) {
        sh 'npm version  ${RELEASE_SCOPE} --no-commit-hooks -m "Publish version %s [ci skip]"'
        sh 'npm publish --access public --ignore-script'
        sh 'git push origin main --no-verify'
        sh 'git push origin --tags --no-verify'
      }
    }
  }
}
```

**Reply by mountHouli** [reply]
+1

Thanks @gh250100 for posting and thanks @ethomson for triaging.

This would be a great feature. My team has a system that has artifacts/data that are identified during our deployment workflow, that we need a human to be able to choose to either delete or archive. It would be great if the workflow could pause to prompt the user what to do with each item discovered.

**Reply by NivEz** [reply]
@mountHouli @gh250100 @elmijo 
I have just released ["Interactive Inputs Action"](https://github.com/marketplace/actions/interactive-inputs-action), a GitHub Action that enables you to prompt users with options on Telegram during the workflow run. You can define parameters such as a timeout for the prompt, default choice, and more.
Feel free to give it a try, I would love to get your feedback 🙂

**Comment by KanakBarfa** [comment]
One way would be to format your comment in such a way so that if you want an yes/no input, then you can use commit message of the form- "::yes:: {Commit message}" and then parse it later, then use if statement in the step to skip it or not accordingly
